### PR TITLE
[URP] Native Render Pass depth resolve support

### PR DESCRIPTION
**Pull request by manuele-bonanno**
### Purpose of this PR
Depth resolve support was recently added to Unity (currently on Metal and Vulkan backends only).
This PR adds depth resolve support when Native Render Pass is enabled.

It also fixes 2 issues:
- some Vulkan CopyDepth pass issues, which were caused by the CopyDepth pass using the resolved depth surface as source texture, even when depth resolve was not done, causing the copied depth texture to be black.
- InvokeOnRenderObjectCallbackPass could potentially cause empty renderpasses which would lead to issues on Vulkan. 

For depth resolve to work on Metal, this trunk PR that addeds depth resolve to the RenderPass API is needed:
https://ono.unity3d.com/unity/unity/pull-request/135156/_/graphics/render_pass_api_depth_resolve

---
### Testing status
local Metal and Vulkan

---

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/Passes/InvokeOnRenderObjectCallbackPass.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@ public InvokeOnRenderObjectCallbackPass(RenderPassEvent evt)
         {
             base.profilingSampler = new ProfilingSampler(nameof(InvokeOnRenderObjectCallbackPass));
             renderPassEvent = evt;
+            useNativeRenderPass = false;
```
Comment (jonuuukas) [inline com.unity.render-pipelines.universal/Runtime/Passes/InvokeOnRenderObjectCallbackPass.cs:13]: we should note this as a TODO, as technically this shouldn't be breaking RenderPass

**Review by jonuuukas — APPROVED** [review]
LGTM

**Review by jonuuukas — APPROVED** [review]

**Review by thomas-zeng — APPROVED** [review]

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs [hunk 2]
```diff
@@ -1181,7 +1206,7 @@ bool CanCopyDepth(ref CameraData cameraData)
             bool msaaDepthResolve = msaaEnabledForCamera && SystemInfo.supportsMultisampledTextures != 0;
 
             // copying depth on GLES3 is giving invalid results. Needs investigation (Fogbugz issue 1339401)
```
Comment (sandy-carter-unity) [inline com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs:1208]: This issue was closed by design. Should we remove the comment "Needs investigation"?

**Review by sandy-carter-unity — CHANGES_REQUESTED** [review]

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs (hunk 2 above)
Comment (sandy-carter-unity) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs:1208]: Maybe "needs driver fix"?

**Review by sandy-carter-unity — APPROVED** [review]
These regressed tests are all fixed with this PR:
* [URP_Foundation on Win_Vulkan_Standalone_mono_Linear](https://unity-ci.cds.internal.unity3d.com/job/10275767)
* [URP_Foundation on Linux_Vulkan_Standalone_mono_Linear](https://unity-ci.cds.internal.unity3d.com/job/10275817)
* [URP_Foundation on Win_DX12_Standalone_mono_Linear](https://unity-ci.cds.internal.unity3d.com/job/10275829)
* [URP_Foundation on Win_DX12_playmode_mono_Linear](https://unity-ci.cds.internal.unity3d.com/job/10275834)
* [URP_Foundation on Android_Vulkan_Standalone_il2cpp_Linear](https://unity-ci.cds.internal.unity3d.com/job/10275837)

File: com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs (hunk 2 above)
Comment (manuele-bonanno) [reply com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs]: changed it here:
https://github.com/Unity-Technologies/Graphics/pull/6217/commits/296526b63828a0529bfa703451730b99b54a50c3

**Review by phi-lira — APPROVED** [review]

**Comment by phi-lira** [comment]
VFX_URP on Win_DX11_playmode_mono_Linear on version trunk  - failing on master 🟡

**Pull request merged**